lib/kontrast/api_client.rb

```diff
@@ -0,0 +1,56 @@
+require 'faraday'
+
+module Kontrast
+    class ApiClient
+
+        attr_reader :responses, :env
+        attr_writer :headers
+
+        def initialize(env, host, app_id, app_secret, headers: {})
+            @env = env
+            @host = host
+            @app_id = app_id
+            @app_secret = app_secret
+            @connection = nil
+            @responses = {}
+            @headers = headers
+        end
+
+        def fetch(path, save_file: false, folder_name: "")
+            response = connection.get("#{path}") do |req|
+                req.headers['Authorization'] = "Bearer #{token}"
+                req.headers.merge!(@headers)
+            end
+            data = JSON.parse(response.body)
```

Comment:
I like the idea of making the tool more generic, but:

- I've seen JSON responses being returned with a wrong content type, like 'text/html'
- Assuming JSON for all responses isn't that crazy as most API are using JSON

anyway, I'd be ok adding support for more content type, but this would need more than just reading the content type, we would need a way to override the parser if the content type is wrong.

I think we should add an issue for that and add it if needs be in a future version.